### PR TITLE
Hybrid straddle

### PR DESCRIPTION
**Pull request by rgbkrk**
Just an idea:
- Make a fast validator based on `ajv`
- Verbose but slow validator based on `jayschema`

/cc @jdfreder

### CODE REVIEW

File: nbformat.js [hunk 1]
```diff
@@ -1,31 +1,92 @@
...
-    this.addSchema(v4Schema, '/v4');
+    JaySchema.call(this);
+
+    this.ajv = new Ajv();
+
+    this.ajv.addSchema(v3Schema, '/v3');
+    this.ajv.addSchema(v4Schema, '/v4');
+
+    this.register(v3Schema, '/v3');
+    this.register(v4Schema, '/v4');
```
Comment (rgbkrk) [inline nbformat.js:20]: If we straddle, we should make sure that `register` gets overridden to call `JaySchema.register.call(this...)` followed by `this.ajv.addSchema...`.

**Comment by jdfreder** [comment]
Something about this seems overly complicated.

**Comment by rgbkrk** [comment]
> Something about this seems overly complicated.

Yup.

**Comment by rgbkrk** [comment]
I think I'd prefer to err on the side of verbosity and reliability. If/when we notice it's a bottleneck we can switch over. At the very least, we should export the schemas themselves so people can validate on their own with their own validator.

**Comment by jdfreder** [comment]
> I think I'd prefer to err on the side of verbosity and reliability.

:+1:

**Comment by rgbkrk** [comment]
Also closing as an old unresolved PR.

**Pull request closed without merging**